src/serialize.h

```diff
@@ -47,87 +48,85 @@ static const unsigned int MAX_VECTOR_ALLOCATE = 5000000;
 struct deserialize_type {};
 constexpr deserialize_type deserialize {};
 
-//! Safely convert odd char pointer types to standard ones.
-inline char* CharCast(char* c) { return c; }
-inline char* CharCast(unsigned char* c) { return (char*)c; }
-inline const char* CharCast(const char* c) { return c; }
-inline const char* CharCast(const unsigned char* c) { return (const char*)c; }
+//! Safely convert a data pointer to a uint8_t data pointer.
+inline uint8_t* Uint8Ptr(void* data) { return reinterpret_cast<uint8_t*>(data); }
+inline const uint8_t* Uint8Ptr(const void* data) { return reinterpret_cast<const uint8_t*>(data); }
 
 /*
  * Lowest-level serialization and conversion.
  * @note Sizes of these types are verified in the tests
  */
 template<typename Stream> inline void ser_writedata8(Stream &s, uint8_t obj)
 {
-    s.write((char*)&obj, 1);
```

Comment:
nit: `Uint8Ptr` unnecessary as this is already `uint8_t*`

Reply:
I think I'll keep this as is for symmetry with the other helpers.